core/drivers/stm32_gpio.c

```diff
@@ -77,6 +80,132 @@ static unsigned int gpio_lock;
 static STAILQ_HEAD(, stm32_gpio_bank) bank_list =
 		STAILQ_HEAD_INITIALIZER(bank_list);
 
+static struct stm32_gpio_bank *gpio_chip_to_bank(struct gpio_chip *chip)
+{
+	return container_of(chip, struct stm32_gpio_bank, gpio_chip);
+}
+
+static enum gpio_level stm32_gpio_get_level(struct gpio_chip *chip,
+					    unsigned int gpio_pin)
+{
+	struct stm32_gpio_bank *bank = gpio_chip_to_bank(chip);
+	enum gpio_level level = GPIO_LEVEL_HIGH;
+	unsigned int reg_offset = 0;
+	unsigned int mode = 0;
+
+	assert(gpio_pin < bank->ngpios);
+	clk_enable(bank->clock);
```

Comment:
Same question here on return code

Reply:
I'll make the function panicking, and consider `TEE_Result` return code for GPIO operation handlers in another P-R.